pkg/solana/chainwriter/chain_writer.go

```diff
@@ -310,10 +389,13 @@ func (s *SolanaChainWriterService) SubmitTransaction(ctx context.Context, contra
 	discriminator := GetDiscriminator(methodConfig.ChainSpecificName)
 	encodedPayload = append(discriminator[:], encodedPayload...)
 
+	// Combine the two sets of instructions into one slice
+	var instructions []solana.Instruction
+	instructions = append(instructions, createATAinstructions...)
```

Comment:
Probably makes sense to split this into 2 separate TXs to save space for the main TX. @amit-momin can we guarantee the order of execution if we do this?

Reply:
Just mentioning this here for visibility before I make the change. I think we may need to poll for finality if we split it. We would broadcast the transactions sequentially but don't think there's any guarantees that they are included in that order. Since they interact with the same account, I think Solana's localized gas markets would mean the tx with the higher price out of the 2 would get included first. Since they would be individual tx in the TXM, I don't think we can make the distinction to ensure the ATA tx always stays higher priced.

Reply:
Ok. Regardless I think it makes sense to split them up since tx sizes are so small on Solana. Should probably check with the on-chain team to confirm they agree